File: year2/semestrul2/MPP/Laboratoare/lab4recuperare/src/main/java/org/example/Model/Inscriere.java
package org.example.Model;

public class Inscriere extends Entity<Long> {
    private String id_inscriere;
    private String id_participant;
    private String concurs_name;
    private String timestamp;

    public Inscriere(String id_inscriere, String id_participant, String concurs_name, String timestamp) {
        this.id_inscriere = id_inscriere;
        this.id_participant = id_participant;
        this.concurs_name = concurs_name;
        this.timestamp = timestamp;
    }

    public String getId_inscriere() {
        return id_inscriere;
    }

    public String getId_participant() {
        return id_participant;
    }

    public String getConcurs_name() {
        return concurs_name;
    }

    public String getTimestamp() {
        return timestamp;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Inscriere)) return false;
        Inscriere i = (Inscriere) o;
        return java.util.Objects.equals(id_inscriere, i.id_inscriere);
    }

    @Override
    public int hashCode() {
        return java.util.Objects.hash(id_inscriere);
    }

    @Override
    public String toString() {
        return "Inscriere{" +
                "id='" + id_inscriere + '\'' +
                ", participant='" + id_participant + '\'' +
                ", concurs='" + concurs_name + '\'' +
                ", timestamp='" + timestamp + '\'' +
                '}';
    }
}

File: year2/semestrul2/MPP/Laboratoare/AvaloniaApplication1/Models/Inscriere.cs
using System;

namespace AvaloniaApplication1.Models
{
    public class Inscriere : Entity<long>
    {
        public string IdInscriere { get; set; }
        public string IdParticipant { get; set; }
        public string ConcursName { get; set; }
        public string Timestamp { get; set; }

        public Inscriere(string idInscriere, string idParticipant, string concursName, string timestamp)
        {
            IdInscriere = idInscriere;
            IdParticipant = idParticipant;
            ConcursName = concursName;
            Timestamp = timestamp;
        }

        public override string ToString()
        {
            return $"Inscriere{{Id='{IdInscriere}', Participant='{IdParticipant}', Concurs='{ConcursName}', Timestamp='{Timestamp}'}}";
        }

        public override bool Equals(object obj)
        {
            return obj is Inscriere inscriere && IdInscriere == inscriere.IdInscriere;
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(IdInscriere);
        }
    }
}